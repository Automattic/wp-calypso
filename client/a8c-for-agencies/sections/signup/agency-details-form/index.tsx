import { Button, Gridicon, FormLabel } from '@automattic/components';
import { useTranslate } from 'i18n-calypso';
import { useCallback, useState, useMemo, ChangeEvent, useEffect } from 'react';
import SearchableDropdown from 'calypso/a8c-for-agencies/components/searchable-dropdown';
import TextPlaceholder from 'calypso/a8c-for-agencies/components/text-placeholder';
import FormFieldset from 'calypso/components/forms/form-fieldset';
import FormSelect from 'calypso/components/forms/form-select';
import FormTextInput from 'calypso/components/forms/form-text-input';
import MultiCheckbox, { ChangeList } from 'calypso/components/forms/multi-checkbox';
import { Option as CountryOption, useCountriesAndStates } from './hooks/use-countries-and-states';
import { AgencyDetailsPayload } from './types';
import type { FormEventHandler } from 'react';

import './style.scss';

function getCountry( country: string, options: CountryOption[] ): string {
	if ( options.length < 1 ) {
		return country;
	}

	for ( let i = 0; i < options.length; i++ ) {
		if ( options[ i ].value === country ) {
			return country;
		}
	}

	return options[ 0 ].value;
}

interface Props {
	includeTermsOfService?: boolean;
	isLoading: boolean;
	onSubmit: ( payload: AgencyDetailsPayload ) => void;
	referer?: string | null;
	initialValues?: {
		agencyName?: string;
		agencyUrl?: string;
		managedSites?: string;
		servicesOffered?: string[];
		city?: string;
		line1?: string;
		line2?: string;
		country?: string;
		postalCode?: string;
		state?: string;
	};
	submitLabel: string;
}

export default function AgencyDetailsForm( {
	includeTermsOfService = false,
	isLoading,
	initialValues = {},
	onSubmit,
	referer,
	submitLabel,
}: Props ) {
	const translate = useTranslate();
	const { countryOptions, stateOptionsMap } = useCountriesAndStates();
	const showCountryFields = countryOptions.length > 0;

	const [ countryValue, setCountryValue ] = useState( initialValues.country ?? '' );
	const [ city, setCity ] = useState( initialValues.city ?? '' );
	const [ line1, setLine1 ] = useState( initialValues.line1 ?? '' );
	const [ line2, setLine2 ] = useState( initialValues.line2 ?? '' );
	const [ postalCode, setPostalCode ] = useState( initialValues.postalCode ?? '' );
	const [ addressState, setAddressState ] = useState( initialValues.state ?? '' );
	const [ agencyName, setAgencyName ] = useState( initialValues.agencyName ?? '' );
	const [ agencyUrl, setAgencyUrl ] = useState( initialValues.agencyUrl ?? '' );
	const [ managedSites, setManagedSites ] = useState( initialValues.managedSites ?? '1-20' );
	const [ servicesOffered, setServicesOffered ] = useState( initialValues.servicesOffered ?? [] );

	const country = getCountry( countryValue, countryOptions );
	const stateOptions = stateOptionsMap[ country ];

	useEffect( () => {
		// Reset the value of state since our options have changed.
		setAddressState( stateOptions?.length ? stateOptions[ 0 ].value : '' );
	}, [ stateOptions ] );

	const payload: AgencyDetailsPayload = useMemo(
		() => ( {
			agencyName,
			agencyUrl,
			managedSites,
			servicesOffered,
			city,
			line1,
			line2,
			country,
			postalCode,
			state: addressState,
			referer,
			...( includeTermsOfService ? { tos: 'consented' } : {} ),
		} ),
		[
			agencyName,
			agencyUrl,
			managedSites,
			servicesOffered,
			city,
			line1,
			line2,
			country,
			postalCode,
			addressState,
			referer,
			includeTermsOfService,
		]
	);

	const handleSubmit = useCallback(
		( e: React.SyntheticEvent ) => {
			e.preventDefault();

			if ( ! showCountryFields || isLoading ) {
				return;
			}

			onSubmit( payload );
		},
		[ showCountryFields, isLoading, onSubmit, payload ]
	);

	const getServicesOfferedOptions = () => {
		return [
			{ value: 'strategy_consulting', label: translate( 'Strategy consulting' ) },
			{ value: 'website_design_development', label: translate( 'Website design & development' ) },
			{ value: 'performance_optimization', label: translate( 'Performance optimization' ) },
			{ value: 'digital_strategy_marketing', label: translate( 'Digital strategy & marketing' ) },
			{ value: 'maintenance_support_plans', label: translate( 'Maintenance & support plans' ) },
		];
	};

	// <FormSelect> complains if we "just" pass "setManagedSites" because it expects
	// React.FormEventHandler, so this wrapper function is made to satisfy everything
	// in an easily readable way.
	const handleSetManagedSites: FormEventHandler = ( { target } ) => {
		const value: string = ( target as HTMLSelectElement ).value;
		setManagedSites( value );
	};

	const handleSetServicesOffered = ( services: ChangeList< string > ) => {
		setServicesOffered( services.value );
	};

	return (
		<div className="agency-details-form">
			<form onSubmit={ handleSubmit }>
				<FormFieldset>
					<FormLabel htmlFor="agencyName">{ translate( 'Agency name' ) }</FormLabel>
					<FormTextInput
						id="agencyName"
						name="agencyName"
						value={ agencyName }
						onChange={ ( event: ChangeEvent< HTMLInputElement > ) =>
							setAgencyName( event.target.value )
						}
						disabled={ isLoading }
					/>
				</FormFieldset>
				<FormFieldset>
					<FormLabel htmlFor="agencyUrl">{ translate( 'Business URL' ) }</FormLabel>
					<FormTextInput
						id="agencyUrl"
						name="agencyUrl"
						value={ agencyUrl }
						onChange={ ( event: ChangeEvent< HTMLInputElement > ) =>
							setAgencyUrl( event.target.value )
						}
						disabled={ isLoading }
					/>
				</FormFieldset>
				<FormFieldset>
					<FormLabel htmlFor="managed_sites">
						{ translate( 'How many sites do you manage?' ) }
					</FormLabel>
					<FormSelect
						name="managed_sites"
						id="managed_sites"
						value={ managedSites }
						onChange={ handleSetManagedSites }
					>
						<option value="1-5">{ translate( '1-5' ) }</option>
						<option value="6-25">{ translate( '6-25' ) }</option>
						<option value="26-50">{ translate( '26-50' ) }</option>
						<option value="51-100">{ translate( '51-100' ) }</option>
						<option value="101-500">{ translate( '101-500' ) }</option>
						<option value="501+">{ translate( '501+' ) }</option>
					</FormSelect>
				</FormFieldset>
				<FormFieldset>
					<FormLabel htmlFor="services_offered">
						{ translate( 'What services do you offer?', {
							comment:
								'Possible values are: "Strategy consulting", "Website design & development", "Performance optimization", "Digital strategy & marketing", "Maintenance & support plans".',
						} ) }
					</FormLabel>
					<MultiCheckbox
						id="services_offered"
						name="services_offered"
						checked={ servicesOffered }
						options={ getServicesOfferedOptions() }
						// // Using 'as any' to bypass TypeScript type checks due to a known and intentional type mismatch between
						// the expected custom event type for `onChange` and the standard event types.
						onChange={ handleSetServicesOffered as any }
					/>
				</FormFieldset>
				<FormFieldset>
					<FormLabel>{ translate( 'Country' ) }</FormLabel>
					{ showCountryFields && (
						<SearchableDropdown
							value={ countryValue }
							onChange={ ( value ) => {
								setCountryValue( value ?? '' );
							} }
							options={ countryOptions }
							disabled={ isLoading }
						/>
					) }

					{ ! showCountryFields && <TextPlaceholder /> }
				</FormFieldset>
				{ showCountryFields && stateOptions && (
					<FormFieldset>
						<FormLabel>{ translate( 'State' ) }</FormLabel>
						<SearchableDropdown
							value={ addressState }
							onChange={ ( value ) => setAddressState( value ?? '' ) }
							options={ stateOptions }
							disabled={ isLoading }
							allowReset={ false }
						/>
					</FormFieldset>
				) }
				<FormFieldset className="agency-details-form__business-address">
					<FormLabel>{ translate( 'Business address' ) }</FormLabel>
					<FormTextInput
						id="line1"
						name="line1"
						placeholder={ translate( 'Street name and house number' ) }
						value={ line1 }
						onChange={ ( event: ChangeEvent< HTMLInputElement > ) =>
							setLine1( event.target.value )
						}
						disabled={ isLoading }
					/>
					<FormTextInput
						id="line2"
						name="line2"
						placeholder={ translate( 'Apartment, floor, suite or unit number' ) }
						value={ line2 }
						onChange={ ( event: ChangeEvent< HTMLInputElement > ) =>
							setLine2( event.target.value )
						}
						disabled={ isLoading }
					/>
				</FormFieldset>
				<FormFieldset>
					<FormLabel htmlFor="city">{ translate( 'City' ) }</FormLabel>
					<FormTextInput
						id="city"
						name="city"
						value={ city }
						onChange={ ( event: ChangeEvent< HTMLInputElement > ) => setCity( event.target.value ) }
						disabled={ isLoading }
					/>
				</FormFieldset>
				<FormFieldset>
					<FormLabel htmlFor="postalCode">{ translate( 'Postal code' ) }</FormLabel>
					<FormTextInput
						id="postalCode"
						name="postalCode"
						value={ postalCode }
						onChange={ ( event: ChangeEvent< HTMLInputElement > ) =>
							setPostalCode( event.target.value )
						}
						disabled={ isLoading }
					/>
				</FormFieldset>
				{ includeTermsOfService && (
					<div className="company-details-form__tos">
						<p>
							{ translate(
								'By clicking ‘Continue’, you agree to the{{break}}{{/break}}{{link}}%(link_text)s{{icon}}{{/icon}}{{/link}}.',
								{
									components: {
										break: <br />,
										link: (
											<a
												href="https://automattic.com/for/agencies/partnership-agreement"
												target="_blank"
												rel="noopener noreferrer"
											></a>
										),
										icon: <Gridicon icon="external" size={ 18 } />,
									},
									args: { link_text: 'Terms of the Automattic for Agencies Partnership Agreement' },
								}
							) }
						</p>
					</div>
				) }
				<div className="company-details-form__controls">
					<Button
						primary
						type="submit"
						className="company-details-form__submit"
						disabled={ ! showCountryFields || isLoading }
						busy={ isLoading }
					>
						{ submitLabel }
					</Button>
				</div>
			</form>
		</div>
	);
}
