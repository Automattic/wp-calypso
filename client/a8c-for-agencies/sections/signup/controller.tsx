import { type Callback } from '@automattic/calypso-router';
import AgencySignUp from './primary/agency-signup';

export const signUpContext: Callback = ( context, next ) => {
	context.primary = <AgencySignUp />;
	next();
};
