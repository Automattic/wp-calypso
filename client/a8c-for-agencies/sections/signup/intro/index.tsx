import { Button } from '@automattic/components';
import { useTranslate } from 'i18n-calypso';
import A8cForAgenciesLogo from './logos/a8c-for-agencies.svg';
import JetpackLogo from './logos/jetpack.svg';
import PressableLogo from './logos/pressable.svg';
import VipLogo from './logos/vip.svg';
import WooLogo from './logos/woo.svg';
import WpcomLogo from './logos/wpcom.svg';

import './style.scss';

/**
 * Signup Introduction Content
 * Information for new users coming from the client plugin.
 */
export default function SignupIntro( { wpAdminUrl }: { wpAdminUrl?: string } ) {
	const translate = useTranslate();
	return (
		<div className="a4a-signup-intro">
			<img src={ A8cForAgenciesLogo } alt={ translate( 'Automattic for Agencies' ) } />
			<h1>{ translate( "You haven't signed up yet!" ) }</h1>
			<p>
				{ translate(
					"Automattic for Agencies will help to streamline your agency's operations and amplify your business with our suite of best-in-class products."
				) }
			</p>
			<div className="a4a-signup-intro__logos">
				<img src={ WpcomLogo } alt={ translate( 'WordPress.com' ) } />
				<img src={ WooLogo } alt={ translate( 'WooCommerce' ) } />
				<img src={ JetpackLogo } alt={ translate( 'Jetpack' ) } />
				<img src={ VipLogo } alt={ translate( 'WordPress VIP' ) } />
				<img src={ PressableLogo } alt={ translate( 'Pressable' ) } />
			</div>
			<h4 className="a4a-signup-intro__separator">{ translate( 'Get Access' ) }</h4>
			<p>
				{ translate(
					"The Automattic for Agencies program is open for sign ups. Manage all of your clients' sites and client relationships in one platform. And more!"
				) }
			</p>
			<Button href="https://automattic.com/for/agencies" primary target="_blank">
				{ translate( 'Visit the program website' ) }
			</Button>
			{ wpAdminUrl && (
				<a href={ wpAdminUrl } className="a4a-signup-intro__wp-admin-link">
					{ translate( 'Back to WP-Admin' ) }
				</a>
			) }
		</div>
	);
}
