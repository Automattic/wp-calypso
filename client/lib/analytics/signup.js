/**
 * External dependencies
 */
import debug from 'debug';

/**
 * Internal dependencies
 */
import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
import { identifyUser } from 'calypso/lib/analytics/identify-user';
import { gaRecordEvent } from 'calypso/lib/analytics/ga';
import { addToQueue } from 'calypso/lib/analytics/queue';
import {
	adTrackSignupStart,
	adTrackSignupComplete,
	adTrackRegistration,
} from 'calypso/lib/analytics/ad-tracking';

const signupDebug = debug( 'calypso:analytics:signup' );

export function recordSignupStart( flow, ref ) {
	// Tracks
	recordTracksEvent( 'calypso_signup_start', { flow, ref } );
	// Google Analytics
	gaRecordEvent( 'Signup', 'calypso_signup_start' );
	// Marketing
	adTrackSignupStart( flow );
}

export function recordSignupComplete(
	{ flow, siteId, isNewUser, hasCartItems, isNew7DUserSite },
	now
) {
	const isNewSite = !! siteId;

	if ( ! now ) {
		// Delay using the analytics localStorage queue.
		return addToQueue(
			'signup',
			'recordSignupComplete',
			{ flow, siteId, isNewUser, hasCartItems, isNew7DUserSite },
			true
		);
	}

	// Tracks
	// Note that Tracks expects blog_id to differntiate sites, hence using
	// blog_id instead of site_id here. We keep using "siteId" otherwise since
	// all the other fields still refer with "site". e.g. isNewSite
	recordTracksEvent( 'calypso_signup_complete', {
		flow,
		blog_id: siteId,
		is_new_user: isNewUser,
		is_new_site: isNewSite,
		has_cart_items: hasCartItems,
	} );

	// Google Analytics
	const flags = [
		isNewUser && 'is_new_user',
		isNewSite && 'is_new_site',
		hasCartItems && 'has_cart_items',
	].filter( Boolean );

	gaRecordEvent( 'Signup', 'calypso_signup_complete:' + flags.join( ',' ) );

	if ( isNew7DUserSite ) {
		// Tracks
		recordTracksEvent( 'calypso_new_user_site_creation', { flow } );

		// Google Analytics
		gaRecordEvent( 'Signup', 'calypso_new_user_site_creation' );
	}

	adTrackSignupComplete( { isNewUserSite: isNewUser && isNewSite } );
}

export function recordSignupStep( flow, step ) {
	recordTracksEvent( 'calypso_signup_step_start', { flow, step } );
}

export function recordSignupInvalidStep( flow, step ) {
	recordTracksEvent( 'calypso_signup_goto_invalid_step', { flow, step } );
}

/**
 * Records registration event.
 *
 * @param {object} param {}
 * @param {object} param.userData User data
 * @param {string} param.userData.ID User id
 * @param {string} param.userData.username Username
 * @param {string} param.userData.email Email
 * @param {string} param.flow Registration flow
 * @param {string} param.type Registration type
 */
export function recordRegistration( { userData, flow, type } ) {
	signupDebug( 'recordRegistration:', { userData, flow, type } );

	// Tracks user identification
	identifyUser( userData );
	// Tracks
	recordTracksEvent( 'calypso_user_registration_complete', { flow, type } );
	// Google Analytics
	gaRecordEvent( 'Signup', 'calypso_user_registration_complete' );
	// Marketing
	adTrackRegistration();
}
