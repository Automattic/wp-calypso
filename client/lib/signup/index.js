export { default as PasswordlessSignupForm } from './passwordless';
export { default as SocialSignupForm } from './social';
