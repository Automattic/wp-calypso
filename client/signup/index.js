/**
 * External dependencies
 */
var page = require( 'page' );

/**
 * Internal dependencies
 */
var controller = require( './controller' ),
	jetpackConnectController = require( './jetpack-connect/controller' ),
	adTracking = require( 'lib/analytics/ad-tracking' ),
	config = require( 'config' );

module.exports = function() {
	if ( config.isEnabled( 'phone_signup' ) ) {
		page( '/phone/:lang?', controller.phoneSignup );
	}

	page(
		'/start/:flowName?/:stepName?/:stepSectionName?/:lang?',
		adTracking.retarget,
		controller.saveRefParameter,
		controller.saveQueryObject,
		controller.redirectWithoutLocaleIfLoggedIn,
		controller.redirectToFlow,
		controller.start
	);

	if ( config.isEnabled( 'login' ) ) {
		page( '/log-in/:lang?', controller.login );
	}

	if ( config.isEnabled( 'jetpack/connect' ) ) {
		page(
			'/jetpack/connect/authorize/:locale?',
			jetpackConnectController.redirectWithoutLocaleifLoggedIn,
			jetpackConnectController.saveQueryObject,
			jetpackConnectController.authorizeForm
		);

		page(
			'/jetpack/connect/:locale?',
			jetpackConnectController.redirectWithoutLocaleifLoggedIn,
			jetpackConnectController.connect
		);
	}

	if ( config.isEnabled( 'jetpack/sso' ) ) {
		page( '/jetpack/sso', jetpackConnectController.saveQueryObject, jetpackConnectController.sso );
	}
};
