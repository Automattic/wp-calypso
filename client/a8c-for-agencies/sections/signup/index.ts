import page from '@automattic/calypso-router';
import { makeLayout, render as clientRender } from 'calypso/controller';
import * as controller from './controller';

export default function () {
	page( '/signup/resume', controller.signUpResumeContext, makeLayout, clientRender );
	page( '/signup', controller.signUpContext, makeLayout, clientRender );
}
