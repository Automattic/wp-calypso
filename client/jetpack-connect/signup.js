/** @format */
/**
 * Handle log in and sign up as part of the Jetpack Connect flow
 *
 * For user creation, this component relies on redux to store state as a user is created via a
 * series of actions. Eventually this results in updating the `authorizationData.userData` prop on
 * this component.
 *
 * When this component receives `userData`, it renders a `<WpcomLoginForm />` with the userData,
 * which handles logging in the new user and redirection.
 */

/**
 * External dependencies
 */
import React, { Component } from 'react';
import debugFactory from 'debug';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { get } from 'lodash';
import { localize } from 'i18n-calypso';

/**
 * Internal dependencies
 */
import { addQueryArgs } from 'lib/route';
import AuthFormHeader from './auth-form-header';
import config from 'config';
import HelpButton from './help-button';
import LocaleSuggestions from 'components/locale-suggestions';
import LoggedOutFormLinkItem from 'components/logged-out-form/link-item';
import LoggedOutFormLinks from 'components/logged-out-form/links';
import MainWrapper from './main-wrapper';
import SignupForm from 'components/signup-form';
import WpcomLoginForm from 'signup/wpcom-login-form';
import { authQueryPropTypes } from './utils';
import { createAccount as createAccountAction } from 'state/jetpack-connect/actions';
import { getAuthorizationData } from 'state/jetpack-connect/selectors';
import { login } from 'lib/paths';
import { recordTracksEvent as recordTracksEventAction } from 'state/analytics/actions';

const debug = debugFactory( 'calypso:jetpack-connect:authorize-form' );

export class JetpackSignup extends Component {
	static propTypes = {
		authQuery: authQueryPropTypes.isRequired,
		locale: PropTypes.string,
		path: PropTypes.string,

		// Connected props
		authorizationData: PropTypes.shape( {
			bearerToken: PropTypes.string,
			isAuthorizing: PropTypes.bool,
			userData: PropTypes.object,
		} ).isRequired,
		createAccount: PropTypes.func.isRequired,
		recordTracksEvent: PropTypes.func.isRequired,
		translate: PropTypes.func.isRequired,
	};

	componentWillMount() {
		const { from, clientId } = this.props.authQuery;
		this.props.recordTracksEvent( 'calypso_jpc_authorize_form_view', {
			from,
			site: clientId,
		} );
	}

	componentDidMount() {
		const { from, clientId } = this.props.authQuery;
		this.props.recordTracksEvent( 'calypso_jpc_signup_view', {
			from,
			site: clientId,
		} );
	}

	handleSubmitSignup = ( form, userData ) => {
		debug( 'submiting new account', form, userData );
		this.props.createAccount( userData );
	};

	handleClickHelp = () => {
		this.props.recordTracksEvent( 'calypso_jpc_help_link_click' );
	};

	/**
	 * Log in the new user
	 *
	 * After an account is created, `authorizationData.userData` is populated
	 * and we render this component to log the new user in.
	 *
	 * @return {Object} React element for render.
	 */
	renderLoginUser() {
		const { userData, bearerToken } = this.props.authorizationData;

		return (
			<WpcomLoginForm
				log={ userData.username }
				authorization={ 'Bearer ' + bearerToken }
				emailAddress={ this.props.authQuery.userEmail }
				redirectTo={ addQueryArgs( { auth_approved: true }, window.location.href ) }
			/>
		);
	}

	renderLocaleSuggestions() {
		return this.props.locale ? (
			<LocaleSuggestions path={ this.props.path } locale={ this.props.locale } />
		) : null;
	}

	renderFooterLink() {
		const emailAddress = this.props.authQuery.userEmail;

		return (
			<LoggedOutFormLinks>
				<LoggedOutFormLinkItem
					href={ login( {
						emailAddress,
						isJetpack: true,
						isNative: config.isEnabled( 'login/native-login-links' ),
						redirectTo: window.location.href,
					} ) }
				>
					{ this.props.translate( 'Already have an account? Sign in' ) }
				</LoggedOutFormLinkItem>
				<HelpButton onClick={ this.handleClickHelp } />
			</LoggedOutFormLinks>
		);
	}

	render() {
		const { isAuthorizing, userData } = this.props.authorizationData;

		return (
			<MainWrapper>
				<div className="jetpack-connect__authorize-form">
					{ this.renderLocaleSuggestions() }
					<AuthFormHeader authQuery={ this.props.authQuery } />
					<SignupForm
						redirectToAfterLoginUrl="!! THIS PROP IS REQUIRED BUT UNUSED !!"
						disabled={ isAuthorizing }
						submitting={ isAuthorizing }
						submitForm={ this.handleSubmitSignup }
						submitButtonText={ this.props.translate( 'Sign Up and Connect Jetpack' ) }
						footerLink={ this.renderFooterLink() }
						email={ this.props.authQuery.userEmail }
						suggestedUsername={ get( userData, 'username', '' ) }
					/>
					{ userData && this.renderLoginUser() }
				</div>
			</MainWrapper>
		);
	}
}

export default connect(
	state => ( {
		authorizationData: getAuthorizationData( state ),
	} ),
	{
		recordTracksEvent: recordTracksEventAction,
		createAccount: createAccountAction,
	}
)( localize( JetpackSignup ) );
