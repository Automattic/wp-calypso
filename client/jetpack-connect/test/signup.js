/**
 * @jest-environment jsdom
 */

/**
 * External dependencies
 */
import deepFreeze from 'deep-freeze';
import React from 'react';
import { identity, noop } from 'lodash';
import { shallow } from 'enzyme';

/**
 * Internal dependencies
 */
import LocaleSuggestions from 'calypso/components/locale-suggestions';
import { JetpackSignup } from '../signup.js';

const CLIENT_ID = 98765;
const SITE_SLUG = 'an.example.site';
const DEFAULT_PROPS = deepFreeze( {
	authorizationData: {
		authorizeError: false,
		authorizeSuccess: false,
		clientId: CLIENT_ID,
		clientNotResponding: true,
		isAuthorizing: false,
		timestamp: 1509368045859,
		userAlreadyConnected: false,
	},
	authQuery: {
		authApproved: false,
		blogname: 'Example Blog',
		clientId: CLIENT_ID,
		from: 'banner-44-slide-1-dashboard',
		homeUrl: `http://${ SITE_SLUG }`,
		jpVersion: '5.4',
		nonce: 'fooBarNonce',
		redirectAfterAuth: `http://${ SITE_SLUG }/wp-admin/admin.php?page=jetpack`,
		redirectUri: `http://${ SITE_SLUG }/wp-admin/admin.php?page=jetpack&action=authorize&_wpnonce=fooBarNonce&redirect=http%3A%2F%2F${ SITE_SLUG }%2Fwp-admin%2Fadmin.php%3Fpage%3Djetpack`,
		scope: 'administrator:fooBarBaz',
		secret: 'fooBarSecret',
		site: `http://${ SITE_SLUG }`,
		siteIcon: '',
		siteUrl: `http://${ SITE_SLUG }`,
		state: '1',
		userEmail: `email@${ SITE_SLUG }`,
	},
	createAccount: noop,
	path: '/jetpack/connect/authorize',
	recordTracksEvent: noop,
	translate: identity,
} );

describe( 'JetpackSignup', () => {
	test( 'should render', () => {
		const wrapper = shallow( <JetpackSignup { ...DEFAULT_PROPS } /> );

		expect( wrapper ).toMatchSnapshot();
	} );

	test( 'should render with locale suggestions', () => {
		const wrapper = shallow(
			<JetpackSignup
				{ ...DEFAULT_PROPS }
				authorizationData={ {
					...DEFAULT_PROPS.authorizationData,
					locale: 'es',
				} }
				locale="es"
				path="/jetpack/connect/authorize/es"
			/>
		);

		expect( wrapper ).toMatchSnapshot();
		expect( wrapper.find( LocaleSuggestions ) ).toHaveLength( 1 );
	} );
} );
