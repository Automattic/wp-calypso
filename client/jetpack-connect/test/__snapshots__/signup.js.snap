// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JetpackSignup should render 1`] = `
<Connect(Localized(JetpackConnectMainWrapper))
  isWoo={false}
>
  <div
    className="jetpack-connect__authorize-form"
  >
    <Connect(Localized(AuthFormHeader))
      authQuery={
        Object {
          "authApproved": false,
          "blogname": "Example Blog",
          "clientId": 98765,
          "from": "banner-44-slide-1-dashboard",
          "homeUrl": "http://an.example.site",
          "jpVersion": "5.4",
          "nonce": "fooBarNonce",
          "redirectAfterAuth": "http://an.example.site/wp-admin/admin.php?page=jetpack",
          "redirectUri": "http://an.example.site/wp-admin/admin.php?page=jetpack&action=authorize&_wpnonce=fooBarNonce&redirect=http%3A%2F%2Fan.example.site%2Fwp-admin%2Fadmin.php%3Fpage%3Djetpack",
          "scope": "administrator:fooBarBaz",
          "secret": "fooBarSecret",
          "site": "http://an.example.site",
          "siteIcon": "",
          "siteUrl": "http://an.example.site",
          "state": "1",
          "userEmail": "email@an.example.site",
        }
      }
      isWoo={false}
    />
    <Connect(Localized(SignupForm))
      disabled={false}
      email="email@an.example.site"
      footerLink={
        <LoggedOutFormLinks>
          <LoggedOutFormLinkItem
            href="/log-in/jetpack?redirect_to=https%3A%2F%2Fexample.com%2F&email_address=email%40an.example.site"
          >
            Already have an account? Sign in
          </LoggedOutFormLinkItem>
          <Connect(Localized(JetpackConnectHelpButton)) />
        </LoggedOutFormLinks>
      }
      handleSocialResponse={[Function]}
      isSocialSignupEnabled={true}
      redirectToAfterLoginUrl="https://example.com/?auth_approved=true"
      submitButtonText="Create your account"
      submitForm={[Function]}
      submitting={false}
      suggestedUsername=""
    />
  </div>
</Connect(Localized(JetpackConnectMainWrapper))>
`;

exports[`JetpackSignup should render with locale suggestions 1`] = `
<Connect(Localized(JetpackConnectMainWrapper))
  isWoo={false}
>
  <div
    className="jetpack-connect__authorize-form"
  >
    <Connect(LocaleSuggestions)
      locale="es"
      path="/jetpack/connect/authorize/es"
    />
    <Connect(Localized(AuthFormHeader))
      authQuery={
        Object {
          "authApproved": false,
          "blogname": "Example Blog",
          "clientId": 98765,
          "from": "banner-44-slide-1-dashboard",
          "homeUrl": "http://an.example.site",
          "jpVersion": "5.4",
          "nonce": "fooBarNonce",
          "redirectAfterAuth": "http://an.example.site/wp-admin/admin.php?page=jetpack",
          "redirectUri": "http://an.example.site/wp-admin/admin.php?page=jetpack&action=authorize&_wpnonce=fooBarNonce&redirect=http%3A%2F%2Fan.example.site%2Fwp-admin%2Fadmin.php%3Fpage%3Djetpack",
          "scope": "administrator:fooBarBaz",
          "secret": "fooBarSecret",
          "site": "http://an.example.site",
          "siteIcon": "",
          "siteUrl": "http://an.example.site",
          "state": "1",
          "userEmail": "email@an.example.site",
        }
      }
      isWoo={false}
    />
    <Connect(Localized(SignupForm))
      disabled={false}
      email="email@an.example.site"
      footerLink={
        <LoggedOutFormLinks>
          <LoggedOutFormLinkItem
            href="/log-in/jetpack/es?redirect_to=https%3A%2F%2Fexample.com%2F&email_address=email%40an.example.site"
          >
            Already have an account? Sign in
          </LoggedOutFormLinkItem>
          <Connect(Localized(JetpackConnectHelpButton)) />
        </LoggedOutFormLinks>
      }
      handleSocialResponse={[Function]}
      isSocialSignupEnabled={true}
      locale="es"
      redirectToAfterLoginUrl="https://example.com/?auth_approved=true"
      submitButtonText="Create your account"
      submitForm={[Function]}
      submitting={false}
      suggestedUsername=""
    />
  </div>
</Connect(Localized(JetpackConnectMainWrapper))>
`;
